update/delta/delta_1.3.7_001_search_opened_valves.sql

```diff
@@ -0,0 +1,207 @@
+alter table qwat_network.network add column if not exists network_id serial ;
+
+/*
+Check if a node is a valve
+Is the parameter _check_if_network_function is true, then we also check if the valve is a network valve
+*/
+CREATE OR REPLACE FUNCTION qwat_network.ft_check_node_is_valve(_id integer, _check_if_network_function boolean)
+ RETURNS boolean
+ LANGUAGE plpgsql
+AS $function$
+declare 
+    count integer := 0;
+    is_network boolean := false;
+begin
+    
+    --raise notice 'vanne %', _id;
+--     if _id = 54387 then
+--         raise notice 'vanne 54387 %', _check_if_network_function;
+--     end if;
+
+    select count(*) into count from qwat_od.valve where id = _id;
+    if count > 0 then
+        if _check_if_network_function then
+            is_network := (select qwat_network.ft_check_valve_is_network(_id));
+            if is_network then
+                --raise notice '---------- vanne réseau';
+                return true;
+            else
+                return false;
+            end if;
+        end if;
+        return true;
+    end if;
+    return false;
+end
+$function$
+;
+
+/*
+Check if a valve is a network valve
+*/
+CREATE OR REPLACE FUNCTION qwat_network.ft_check_valve_is_network(_id integer)
+ RETURNS boolean
+ LANGUAGE plpgsql
+AS $function$
+declare 
+    _function integer;
+begin
+    select fk_valve_function into _function from qwat_od.valve where id = _id;
+    if _function != 6105 and _function != 6108 then
```

Comment:
instead of hardcoding the keys, shouldn't we rely on a (new?) column of the value list table (valve function)?
otherwise the code is not very resilient.